ansible/roles/cluster_infra/templates/resources.tf.j2

```diff
@@ -386,6 +399,8 @@ resource "openstack_compute_instance_v2" "login" {
         ansible_init_coll_{{ loop.index0 }}_source = "{{ collection.source }}"
       {% endif %}
     {% endfor %} 
+    k3s_server = [for n in openstack_compute_instance_v2.control["control"].network: n.fixed_ip_v4 if n.access_network][0]
```

Comment:
re. using `openstack_compute_instance_v2.control["control"]`, see comments below about changes to caas resources

Reply:
I was using the same method to get the control node's IP used as in the changes added in https://github.com/stackhpc/ansible-slurm-appliance/pull/442 to the cookiecutter environment, which require indexed control nodes, otherwise fails with `Because openstack_compute_instance_v2.control does not have "count" or "for_each" set, references to it must not include an index key. Remove the bracketed index to refer to the single instance of this resource.` Could use 
```suggestion
    k3s_server = openstack_compute_instance_v2.control.network[0].fixed_ip_v4
```
instead but probably nice to keep them consistent

Reply:
- We don't need to keep caas and cookiecutter  consistent if its already different
- caas definitely shouldn't get a set of control nodes (that was a hang-over in cookiecutter from other functionality)
-  I suspect using `openstack_compute_instance_v2.control.network[0].fixed_ip_v4` is fine - it only has 1x network.